### PR TITLE
Refactor filetypeFromNumber in stdlib/File

### LINKED ISSUE #337: Refactor giant if/else in File stdlib to use pattern matching
This is the code in question: https://github.com/grain-lang/grain/blob/4c03b34094e500fb90f7ed81f19fcfddf4c112ce/stdlib/sys/file.gr#L165-L176

Now that we've got `match` guards and matching on constants, we can finally make this a pattern match! 🎉

### PR DESCRIPTION
**Pull request by fa7ad**
Refactor out large if-else chain to pattern matching

Fixes #337

### CODE REVIEW

File: stdlib/sys/file.gr [hunk 1]
```diff
@@ -163,16 +163,17 @@ data DirectoryEntry = {
...
-    if (filetype == 1) BlockDevice else
-    if (filetype == 2) CharacterDevice else
-    if (filetype == 3) Directory else
-    if (filetype == 4) RegularFile else
-    if (filetype == 5) SocketDatagram else
-    if (filetype == 6) SocketStream else
-    if (filetype == 7) SymbolicLink else
-    fail "Unknown filetype"
+  match (filetype) {
+    0 => Unknown
```
Comment (ospencer) [inline stdlib/sys/file.gr:167]: I think this is actually a syntax error right now 🙈 
```suggestion
    | 0 => Unknown
```

**Review by ospencer — CHANGES_REQUESTED** [review]
One small thing, but otherwise looks great! 🎉

**Comment by ospencer** [comment]
Made https://github.com/grain-lang/grain/issues/347 to address that syntax error issue.

**Review by ospencer — APPROVED** [review]
Sweet, thank you! 🎉

**Comment by ospencer** [comment]
Sorry for making you change the code for no reason 🤦

**Comment by fa7ad** [comment]
Hey, don't worry about it, I literally had to click a button on github lol 😅

**Pull request merged**